src/main/java/uk/ac/cam/cl/dtg/segue/api/services/ContentService.java

```diff
@@ -126,7 +127,7 @@ public ContentService(final IContentManager contentManager, @Named(CONTENT_INDEX
                 fieldsToMatchOutput.add(new IContentManager.BooleanSearchClause(
                         pair.getKey(), Constants.BooleanOperator.OR, pair.getValue()));
 
-            } else if (pair.getKey().equals(TYPE_FIELDNAME) && pair.getValue().size() > 1) {
+            } else if ((pair.getKey().equals(TYPE_FIELDNAME) || pair.getKey().equals(TAGS_FIELDNAME)) && pair.getValue().size() > 1) {
```

Comment:
All of the uses in the events facade don't matter in that case - only 2/3 of them use the tags field, and as you say types of query aren't used yet. Also none of the docstrings describe how each field is used anyway (whether they are unions or intersections) so it feels arbitrary to explain how the tag field is turned into a BooleanSearchClause without explaining the other fields as well. I'm happy to do that if needed!

Reply:
The uses in the pages facade are fine given an additional function (that I feel should be available anyway?):
 - The `/pages/questions` endpoint is only used in the `QuestionSearchModal` for the gameboard builder. When you filter by tags in this context, you would probably want 'all' rather than 'any' behaviour. I've added another function to deal with this which is more flexible in how you can build search queries.
 - The `pages/concepts` endpoint is only used in one other place (apart from the CS question filter) in the front-end, and tags aren't used there so this doesn't change anything.

Reply:
With the addition of the new helper function, this actually no longer matters and I can change `getDefaultFieldsToMatch` back to how it was.